tests/e2e/tests/api.spec.js

```diff
@@ -0,0 +1,149 @@
+const { test, expect } = require('@playwright/test');
+
+test.describe('API Endpoints E2E', () => {
+  const baseURL = process.env.BASE_URL || 'http://localhost:8080';
+
+  test('should get weather data for valid city', async ({ request }) => {
+
+    const response = await request.get(`${baseURL}/api/weather?city=London`);
+    
+    if (response.status() === 200) {
+      const weatherData = await response.json();
+      
+      expect(weatherData).toHaveProperty('city');
+      expect(weatherData).toHaveProperty('temperature');
+      expect(weatherData).toHaveProperty('humidity');
+      expect(weatherData).toHaveProperty('description');
+      
+      expect(typeof weatherData.temperature).toBe('number');
+      expect(typeof weatherData.humidity).toBe('number');
+      expect(typeof weatherData.city).toBe('string');
+      expect(typeof weatherData.description).toBe('string');
+    } else {
+
+      expect(response.status()).toBe(500);
+      const errorData = await response.json();
+      expect(errorData).toHaveProperty('error');
+    }
+  });
+
+  test('should return error for missing city parameter', async ({ request }) => {
```

Comment:
It's not call to action, just my opinion. 
These king of tests should not be run by a frontend framework unless you want to test your UI behavior (handling these errors etc).
This test should be covered with your go code, it will run 100 times faster